Superset behaviours of matmul

JLU-Neal opened the pull request.

Ishticode reviewed (changes requested):
This looks right 🙂 but could you also add the transpose parameters in the `test_matmul`

JLU-Neal commented:
Hey @Ishticode !
I did not intend to ask you to review after the first commit but ivy-seed automatically requested you to review hahaha!
Please review again :D 
I have added the transpose parameters and fixed some bugs! Thanks!

Ishticode commented on ivy/functional/backends/numpy/linear_algebra.py:
```diff
@@ -147,10 +147,13 @@ def matmul(
+    except ValueError:
+        print(x1.shape, x2.shape)
     if len(x1.shape) == len(x2.shape) == 1:
         ret = np.array(ret)
```
the second `ret` can be undefined in case both x1 and x2 were were 1D.

JLU-Neal commented:
@Ishticode Hey I want to push the latest commits but seems I have no more write access now! Could you give me the write access for a little while so that I could wrap up this?

Ishticode commented:
> @Ishticode Hey I want to push the latest commits but seems I have no more write access now! Could you give me the write access for a little while so that I could wrap up this?



> @Ishticode Hey I want to push the latest commits but seems I have no more write access now! Could you give me the write access for a little while so that I could wrap up this?

Hmm I m not sure. I will ask @VedPatwardhan and see what we can do.

Ishticode commented:
Hi. I've been told the write permission is strictly for the members currently on team. If its not too much to ask, please copy over the content to a new file from the branch of your own fork. This will allow you to edit on the go and will also save you the trouble of resolving conflicts above. Sorry for inconvenience and thanks for your patience.

ivy-seed commented:
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

ivy-seed commented:
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

JLU-Neal commented:
@Ishticode Hey, sorry about the late response. I tried to do as you said but when I tried to fix the merge conflict, it seems that the superset behaviour of the matmul has already been implemented. Let me know if there is anything else I can help with.
